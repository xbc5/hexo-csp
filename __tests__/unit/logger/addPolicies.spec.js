const logger = require("../../../lib/logger");
const Policies = require("../../../lib/Policies");
const Config = require("../../../lib/config");

it("should should be defined", async () => {
  expect(logger.addPolicies).toBeDefined();
});

function fixture({ env }) {
  const loggerConf = {
    prod: {
      host: "https://production.com",
      path: "/csp-logger",
    },
    dev: {
      host: "localhost",
      port: "4000",
      path: "/csp-logger",
    },
  };
  const config = new Config({ env, logger: loggerConf });
  const policies = new Policies({ env });
  const add = (p = policies, c = config) => {
    return logger.addPolicies(p, c);
  };
  return { config, policies, addPolicies: add };
}
["prod", "dev"].forEach((env) => {
  describe(`for the ${env} environment`, () => {
    it(`should add the ${env} policy`, async () => {
      const { policies, addPolicies } = fixture({ env });
      addPolicies();
      expect(policies.policies).toMatchSnapshot();
    });
  });
});
