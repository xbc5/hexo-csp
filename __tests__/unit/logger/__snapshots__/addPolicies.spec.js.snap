// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for the dev environment should add the dev policy 1`] = `
Array [
  Object {
    "directives": Object {
      "report-uri": Array [
        "localhost:4000/csp-logger",
      ],
    },
    "pattern": ".+",
  },
]
`;

exports[`for the prod environment should add the prod policy 1`] = `
Array [
  Object {
    "directives": Object {
      "report-uri": Array [
        "https://production.com/csp-logger",
      ],
    },
    "pattern": ".+",
  },
]
`;
